x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,728 @@
+    - same logic as for `_bulk_action` with the `export` action explained above: no migration will be performed, but will allow exporting of prebuilt rules
+- **Coverage Overview** - `/rules_coverage_overview` (Internal): This endpoint is called to retrieve the data that populates the MITRE Coverage Overview table, and currently depends on the `immutable` field. It should be updated to read the `prebuilt` field. Since we have to assume that the user will have a mixture of migrated and non-migrated rules, we will have to update the handler to read the `prebuilt` field, with a fallback on the legacy `immutable` field.
+- **(LEGACY) Get Prebuilt Rules and Timeline Status** - `/rules/prepackaged/_status`: currently depends on rules `alert.attributes.params.immutable` to fetch number of custom rules and number of prebuilt rules. Adapt logic to rely on new `alert.attributes.params.prebuilt` field, with fallback to the original, for backwards compatibility.
+- **Get Prebuilt Rules Status** - `GET /prebuilt_rules/status` (Internal): Uses `IPrebuiltRuleObjectsClient` to retrieve instances of prebuilt rules according to the `immutable` field. Adapt logic to rely on new `alert.attributes.params.prebuilt` field, with fallback to the original, for backwards compatibility.
+- **(LEGACY) Install Prebuilt Rules And Timelines** - `PUT /rules/prepackaged`: currently depends on rules `alert.attributes.params.immutable` to fetch custom rules and prebuilt rules. Adapt logic to rely on new `alert.attributes.params.prebuilt` field, with fallback to the original `immutable`, for backwards compatibility.
+- **Installation and Upgrade `_review` and `_perform` endpoints:**
+    - All four endpoints use `IPrebuiltRuleObjectsClient` to retrieve instances of prebuilt rules according to the `immutable` field. Adapt logic to rely on new `alert.attributes.params.prebuilt` field, with fallback to the original, for backwards compatibility.
+        - **Review Rule Installation** - `POST /prebuilt_rules/installation/_review` (Internal):
+            - Uses `convertPrebuiltRuleAssetToRuleResponse`, where we need to set new `prebuilt` field to `true` and `customized` to `false`.
+        - **Perform Rule Installation** - `POST /prebuilt_rules/installation/_install` (Internal):
+            - When new prebuilt rules are created, pass in new `prebuilt` param set to `true`.
+        - **Review Rule Upgrade** - `POST /prebuilt_rules/upgrade/_review` (Internal)
+            - Uses `convertPrebuiltRuleAssetToRuleResponse`, where we need to set new `prebuilt` field to `true` and `customized` to `false`.
+            - This endpoint will need further changes, detailed further down, out of the scope of SO and rule schema updates.
+        - **Perform Rule Upgrade** - `POST /prebuilt_rules/upgrade/_perform` (Internal)
+            - Should update the `upgradeRule` util to include new `prebuilt` and `customized` fields.
```

Comment:
Missing an explanation how and based on what these endpoints would calculate the value of the `customized` field. Mind that there can be cases where non-customized prebuilt rules can become customized after an upgrade, by the upgrade itself, if the user changes the values of some fields manually on upgrade.

Reply:
Will make this explanation where I detail the changes needed for this endpoint for allowing customized fields, as they are strongly related. Left a note about that.